minicluster/src/main/java/org/apache/accumulo/miniclusterImpl/MiniAccumuloClusterImpl.java

```diff
@@ -600,6 +602,97 @@ public synchronized void start() throws IOException, InterruptedException {
     if (executor == null) {
       executor = Executors.newSingleThreadExecutor();
     }
+
+    verifyUp();
+
+  }
+
+  private void verifyUp() throws InterruptedException, IOException {
+
+    Objects.requireNonNull(getClusterControl().managerProcess, "Error starting Manager");
+    Objects.requireNonNull(getClusterControl().managerProcess.info().startInstant().get(),
+        "Error starting Manager");
+
+    Objects.requireNonNull(getClusterControl().gcProcess, "Error starting GC");
+    Objects.requireNonNull(getClusterControl().gcProcess.info().startInstant().get(),
+        "Error starting GC");
+
+    int tsExpectedCount = 0;
+    for (Process tsp : getClusterControl().tabletServerProcesses) {
+      tsExpectedCount++;
+      Objects.requireNonNull(tsp, "Error starting TabletServer");
+      Objects.requireNonNull(tsp.info().startInstant().get(), "Error starting Manager");
+    }
+
+    String zk = config.getExistingZooKeepers() != null ? config.getExistingZooKeepers()
+        : config.getZooKeepers();
+
+    ZooReaderWriter zrw =
+        new ZooReaderWriter(zk, 30000, Property.INSTANCE_SECRET.getDefaultValue());
+
+    String instanceId = null;
+    try {
+      for (String name : zrw.getChildren(Constants.ZROOT + Constants.ZINSTANCES)) {
+        if (name.equals(config.getInstanceName())) {
+          String instanceNamePath = Constants.ZROOT + Constants.ZINSTANCES + "/" + name;
+          byte[] bytes = zrw.getData(instanceNamePath);
+          instanceId = new String(bytes, UTF_8);
+        }
+      }
+    } catch (KeeperException e) {
+      throw new RuntimeException("Unable to read instance name from zookeeper.", e);
+    }
+    String rootPath = ZooUtil.getRoot(instanceId);
+
+    int tsActualCount = 0;
+    int tryCount = 0;
+    while (tsActualCount != tsExpectedCount) {
+      tryCount++;
+      try {
+        tsActualCount = 0;
+        for (String child : zrw.getChildren(rootPath + Constants.ZTSERVERS)) {
+          tsActualCount++;
+          if (zrw.getChildren(rootPath + Constants.ZTSERVERS + "/" + child).isEmpty())
+            throw new RuntimeException("TServer not present in ZooKeeper");
+        }
+      } catch (KeeperException e) {
+        throw new RuntimeException("Unable to read TServer information from zookeeper.", e);
+      }
+      if (tryCount == 10) {
+        throw new RuntimeException("Timed out waiting for TServer information in ZooKeeper");
+      }
+      Thread.sleep(1000);
+    }
+
+    Preconditions.checkArgument(tsExpectedCount == tsActualCount,
```

Comment:
I think this will always be true according to the while loop condition above.

Reply:
Addressed in 35efd1c